src/main/java/org/kohsuke/groovy/sandbox/ScopeTrackingClassCodeExpressionTransformer.java

```diff
@@ -157,15 +157,9 @@ public void visitCatchStatement(CatchStatement statement) {
 
     @Override
     public void visitWhileLoop(WhileStatement loop) {
+        loop.setBooleanExpression((BooleanExpression) transform(loop.getBooleanExpression()));
         try (StackVariableSet scope = new StackVariableSet(this)) {
-            super.visitWhileLoop(loop);
-        }
-    }
-
-    @Override
-    public void visitClosureExpression(ClosureExpression expression) {
-        try (StackVariableSet scope = new StackVariableSet(this)) {
-            super.visitClosureExpression(expression);
+            loop.getLoopBlock().visit(this);
```

Comment:
We do not want to call `super.visitWhileLoop` for the same reason as mentioned above with `visitForLoop`.